pennylane_lightning_gpu/src/simulator/StateVectorCudaMPI.hpp

```diff
@@ -983,6 +1025,188 @@ class StateVectorCudaMPI
+     * global wires, and there are sufficient local wires available for bit swap
+     * operations. 3. Some target wires are located at the global wires, and
+     * there are insufficient local wires available for bit swap operations. For
+     * the first scenario, the `expvalOnPauliBasis` method can be called
+     * directly to obtain the `expval`. In the second scenario, bit swap
+     * operations are necessary before and after the `expvalOnPauliBasis`
+     * operation to get the `expval`. In the third scenario, a temporary state
+     * vector is created to calculate the `bra`, and then the inner product is
+     * used to calculate the `expval`. This function here will check the
+     * corresponding target wires of all word first. If all target wires for
+     * each word are local, the `expvalOnPauliBasis` will be called just once.
+     * Otherwise, each word will be looped and corresponding expval calculation
+     * methods will be adopted.
+     *
+     * @param pauli_words Vector of Pauli-words to evaluate expectation value.
+     * @param tgts Coupled qubit index to apply each Pauli term.
+     * @param coeffs Numpy array buffer of size |pauli_words|
+     * @return auto Expectation value.
+     */
+    auto getExpectationValuePauliWords(
+        const std::vector<std::string> &pauli_words,
+        const std::vector<std::vector<std::size_t>> &tgts,
+        const std::complex<Precision> *coeffs) {
+
+        enum WiresSwapStatus : std::size_t { Local, Swappable, UnSwappable };
+
+        std::vector<double> expect_local(pauli_words.size());
+
+        std::vector<std::size_t> tgtsSwapStatus;
+        std::vector<std::vector<int2>> tgtswirePairs;
+        std::vector<std::vector<size_t>> tgtsInC;
+        tgtsInC.reserve(tgts.size());
+
+        for (const auto &vec : tgts) {
+            std::vector<size_t> tmpVecInt(
+                vec.size()); // Reserve memory for efficiency
+
+            std::transform(vec.begin(), vec.end(), tmpVecInt.begin(),
+                           [&](std::size_t x) {
+                               return this->getTotalNumQubits() - 1 - x;
+                           });
+            tgtsInC.push_back(std::move(tmpVecInt));
+        }
+
+        std::vector<std::vector<size_t>> localTgtsInC;
+        localTgtsInC.reserve(tgts.size());
+
+        for (const auto &vec : tgtsInC) {
+            std::vector<int> statusWires(this->getTotalNumQubits(),
+                                         WireStatus::Default);
+
+            for (size_t i = 0; i < vec.size(); i++) {
+                statusWires[vec[i]] = WireStatus::Target;
+            }
+            size_t StatusGlobalWires =
+                std::reduce(statusWires.begin() + this->getNumLocalQubits(),
+                            statusWires.end());
+
+            if (!StatusGlobalWires) {
+                tgtsSwapStatus.push_back(WiresSwapStatus::Local);
+                localTgtsInC.push_back(vec);
+            } else {
```

Comment:
```suggestion
            if (!StatusGlobalWires) {
                tgtsSwapStatus.push_back(WiresSwapStatus::Local);
                localTgtsInC.push_back(vec);
                continue;
            }
```
Then unindent block below.

Reply:
I would like to keep the current implementation since continue could be not safe.